iceoryx2-bb/cxx/include/iox2/container/static_vector.hpp

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#ifndef INCLUDE_GUARD_IOX2_CONTAINER_STATIC_VECTOR_HPP
+#define INCLUDE_GUARD_IOX2_CONTAINER_STATIC_VECTOR_HPP
+
+#include "iox2/container/config.hpp"
+#include "iox2/container/optional.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <memory>
+#include <type_traits>
+
+namespace iox2 {
+namespace container {
+
+namespace detail {
+    template<typename T, uint64_t N>
+    struct StaticVectorStorage {
+        static_assert(std::is_standard_layout<T>::value, "Storage is only valid for standard layout types.");
+
+        alignas(T) char bytes[sizeof(T) * N];
+        uint64_t size;
+
+        constexpr StaticVectorStorage() noexcept
+        :bytes{}, size(0)
+        {}
+
+        template<size_t M, typename std::enable_if<(N >= M), bool>::type = true>
```

Comment:
This is really nice and something we didn't have for the classic iceoryx version.